app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,28 @@
+class CommentsController < ApplicationController
+  def new
+    @post = Post.find(params[:post_id])
+    @user = User.find(params[:user_id])
+    @comment = Comment.new
```

Comment:
This is the main code. The other above it that are not being used in this method can be removed. To avoid unnecessary database query.